src/lib/pk_pad/emsa_raw/emsa_raw.cpp

```diff
@@ -44,27 +44,27 @@ std::vector<uint8_t> EMSA_Raw::raw_data() {
 /*
 * EMSA-Raw Encode Operation
 */
-std::vector<uint8_t> EMSA_Raw::encoding_of(const std::vector<uint8_t>& msg,
+std::vector<uint8_t> EMSA_Raw::encoding_of(std::span<const uint8_t> msg,
                                            size_t /*output_bits*/,
                                            RandomNumberGenerator& /*rng*/) {
    if(m_expected_size && msg.size() != m_expected_size) {
       throw Invalid_Argument("EMSA_Raw was configured to use a " + std::to_string(m_expected_size) +
                              " byte hash but instead was used for a " + std::to_string(msg.size()) + " hash");
    }
 
-   return msg;
+   return std::vector<uint8_t>(msg.begin(), msg.end());
 }
 
 /*
 * EMSA-Raw Verify Operation
 */
-bool EMSA_Raw::verify(const std::vector<uint8_t>& coded, const std::vector<uint8_t>& raw, size_t /*key_bits*/) {
+bool EMSA_Raw::verify(std::span<const uint8_t> coded, std::span<const uint8_t> raw, size_t /*key_bits*/) {
    if(m_expected_size && raw.size() != m_expected_size) {
       return false;
    }
 
    if(coded.size() == raw.size()) {
-      return (coded == raw);
+      return constant_time_compare(coded, raw);
    }
```

Comment:
Do we even need to perform the length check explicitly?

Reply:
No we do not, but if the lengths are not equal we perform other logic instead, relating to skipping leading zero bytes in the recovered sig.

Reply:
Ah but this can be cleaned up significantly